base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+class BaseDecorator < Nameable
+  def initialze(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  attr_accessor :nameable
+
```

Comment:
- [ ] [Optional] There is a typo error please consider it `initialize` not `initialze`. This is the only thing so I am providing it an optional please consider it.